### PR TITLE
recursion and return statements

### PR DESCRIPTION
**Pull request by Pavel-Durov**
+ function  termination at the return statement
+ function recursion tests

Examples:

```
fun f (){ 
    if (1 < 2){
        return true;
    }
    return false; // should not be executed
}

println(f());

```
```
fun a (){ 
    return true;
    return false; // should not be executed
}
a();
```

```
fun add_10 (x){ 
    if (x < 10) {
        return add_10(x + 1);
    }
    return x;
}
add_10(1);
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -289,19 +300,29 @@ impl Calc {
...
                     }
                 }
             }
         }
+        let result;
         if self.stack.is_empty() {
-            return Ok(None);
+            result = Ok(None);
+        } else {
+            result = Ok(Some(EvalResult::Value(self.stack.pop().unwrap())));
```
Comment (vext01) [inline src/lib.rs:323]: do we know this unwrap can't fail?

Comment (Pavel-Durov) [reply src/lib.rs:323]: I'm not sure cause its been there for a while. I'll add a check for it

Comment (Pavel-Durov) [reply src/lib.rs]: That's actually why I added the `.pop` wrapper function.  Changed :point_right: 85a8e361e54a404ab3c4ccc095538361715eee72

**Comment by Pavel-Durov** [comment]
bors try

**Comment by Pavel-Durov** [comment]
bors try

**Comment by vext01** [comment]
please squash

**Comment by Pavel-Durov** [comment]
squashed :point_right: f5dea2ce4b166b1b61db4ba27bc0b04ef8d8c127

**Pull request merged**